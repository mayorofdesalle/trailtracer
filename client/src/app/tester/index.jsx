import '@testing-library/jest-dom';

import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import './mocks/i18n.mock';
import './mocks/matchMedia.mock';
import './mocks/resizeObserver.mock';

import store from '@app/store';
import GlobalStyle from '@components/style/GlobalStyle';
import Theme from '@components/style/Theme';

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                <GlobalStyle />
                <BrowserRouter>
                    {children}
                </BrowserRouter>
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };
