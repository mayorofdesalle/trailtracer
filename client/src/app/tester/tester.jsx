import { render } from '@testing-library/react'
import { Provider } from 'react-redux';

import './mocks/matchMedia.mock';
import store from '@app/store';
import Theme from '@components/style/Theme'

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                {children}
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };