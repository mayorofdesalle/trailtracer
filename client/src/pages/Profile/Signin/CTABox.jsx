import { useContext } from 'react';
import styled from 'styled-components';

import { slideFromBottom } from '@components/style/anims';
import { pseudoBentoArrowTop } from '@components/ui/Bento/Arrows';
import BentoBox from '@components/ui/Bento/BentoBox';
import Button from '@components/ui/Button';
import Text from '@components/ui/Text';
import PageContext from '@context/PageContext';

/**
 * CTABox
 * @description
 * Styled bento box to contain the CTA content.
 **/
const CTABoxInner = styled(BentoBox)`
    grid-area: CTABox;
    flex-direction: column;
    justify-content: space-around;

    --arrowColor: ${({ theme }) => theme.colors.secondary};
    ${pseudoBentoArrowTop}

    &::before {
        animation: ${slideFromBottom} 1s ease-in-out;
    }

    & > ${Button} {
        width: 80%;
        padding: 0.25rem;
    }

    & > #explanation {
        width: 80%;
    }
`;

const CTABox = () => {
    const context = useContext(PageContext);
    const theme = context.theme;
    const t = context.t;
    const largeScreen = context.largeScreen;

    return (
        <CTABoxInner $glass>
            <Text $color={theme.colors.primary} $heading $ratio={!largeScreen && 0.8}>{t('signinPage.signupTitle')}</Text>
            {largeScreen && (
                <Text id='explanation' $color={theme.colors.text}>
                    {t('signinPage.explanation')}
                </Text>
            )}
            <Button $bgColor={theme.colors.secondary}>
                <Text $color={theme.colors.text} $heading>{t('signinPage.signup')}</Text>
            </Button>
        </CTABoxInner>
    );
};

export default CTABox;