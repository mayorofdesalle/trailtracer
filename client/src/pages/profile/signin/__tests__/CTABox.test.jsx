import { describe, expect, it, render, screen } from '@app/tester';

import CTABox from '../CTABox';

describe('CTABox', () => {

    it('should render correctly', () => {
        render(<CTABox />);
        const ctaBox = screen.getByRole('complementary');
        expect(ctaBox).toBeInTheDocument();
        expect(ctaBox).toHaveStyle({ gridArea: 'CTABox' });

        const button = screen.getByRole('button');
        expect(button).toBeInTheDocument();
    });

});