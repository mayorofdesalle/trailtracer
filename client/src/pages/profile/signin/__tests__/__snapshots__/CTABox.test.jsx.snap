// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CTABox > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-guDLey sc-dmyCSP sc-hIPBNq kHwwsn dHdhmG ezvXUo"
    role="complementary"
  >
    <div
      class="sc-gLLuof kEDizj"
    >
      signinPage.signupTitle
    </div>
    <div
      class="sc-gLLuof fyoZns explanation"
    >
      signinPage.explanation
    </div>
    <button
      class="sc-beySPh fJXTBv"
      type="button"
    >
      <div
        class="sc-gLLuof jqFQvo"
      >
        signinPage.signup
      </div>
    </button>
  </div>
</DocumentFragment>
`;
